src/abstractdataframe/join.jl

```diff
@@ -247,6 +273,11 @@ Join two `DataFrame` objects
   if `true`, duplicate names will be suffixed with `_i`
   (`i` starting at 1 for the first duplicate).
 
+* `check_uniqueness` : checks for whether merge keys are unique before merging.
```

Comment:
I suggest something like (feel free to improve it if you have ideas):
"whether to check that columns passed as the `on` argument define unique keys in each input data frame. Can be a tuple or a pair, with the first element indicating whether to run check for `df1` and the second element for `df2`. By default no check is performed."

Also the name of the argument isn't great, as we already have `makeunique` which refers to column names. Not sure what's the best name. Maybe `validate` as in Pandas? I agree it's not very specific.